content/docs/user-guide/programming/memory-management.md

```diff
@@ -10,72 +10,17 @@ When managing memory in O3DE, use AZ memory management calls and avoid static va
-        static AZStd::vector<ColorB> colors;
-        //...
-    }
-}
-```
-
-By default, the `AZStd::vector` class uses the `SystemAllocator`. Invoking this function on application shutdown, when `SystemAllocator` no longer exists, causes a deletion from a destroyed heap.
-
-### Global Static Variable Example 
-
-On startup, the following code attempts to create a global static variable whose constructor connects to an EBus:
-
-**Example**
-
-```
-// Example of a global static variable that connects to an EBus.
-namespace LmbrCentral
-{
-    // Wraps an IMaterial pointer so that BehaviorContext can use it.
-    class MaterialHandle
-        : public AZ::RenderNotificationsBus::Handler
-    {
-    public:
-        AZ_CLASS_ALLOCATOR(MaterialHandle, AZ::SystemAllocator, 0);
-        AZ_TYPE_INFO(MaterialHandle, "{BF659DC6-ACDD-4062-A52E-4EC053286F4F}");
-        MaterialHandle()
-        {
-            AZ::RenderNotificationsBus::Handler::BusConnect();
-        }
-//...
-// Later in the code, a MaterialHandle in global space is declared:
-MaterialHandle g_defaultMaterialHandle;
-```
-
-The code crashes the engine as soon as the module that contains the global variable definition is loaded. The constructor of the global variable attempts to connect to the EBus, but the EBus system is not ready because it uses an environment variable from a module that has not yet been initialized.
+O3DE initializes its memory system via storing an instances of Allocators in an AZ Environment variables the first time a call to `AZ::AllocatorInstance<AllocatorType>::Get()` is called in [code](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Memory/AllocatorInstance.h#L73-L86).  
+To ensure that Allocators are available in static initialization and static de-initialization of the running applicaition and any dynamically loaded Gem modules, the allocator AZ Environment variables are stored in the `O3DEKernel` shared library which is a link library dependency of the [AzCore Target](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/CMakeLists.txt#L60-L98).  
+As the `AzCore` static library target is a dependency by Gems and O3DE applications, they all statically link the `O3DEKernel` target due to that dependency.  
+On initial load of the application, the `O3DEKernel` library is the first shared library loaded.  
+As loading of statically linked shared libraries occur before static initialization and unloading occurs after static de-initialization of the application, this ensures that the O3DE Allocators are available for the entire duration of the application.  
+The [AZ::Environnment](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Environment.h#L114-L115) is accessed through the O3DEKernel via a DLL interface to enforce the only instance to be available in `O3DEKernel.dll` (Windows), `libO3DEKernel.so` (Linux/Android) or `libO3DEKernel.dylib` (MacOS/iOS).  
```

Comment:
As translated from ancient O3DE texts:
```
One AZ::Environment API to rule them all, 
One AZ::Environment API to find them, 
One AZ::Environment API to bring them all, 
and in the O3DEKernal bind them.
```

```suggestion
Only one [AZ::Environnment](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Environment.h#L114-L115) instance exists and is accessed through the O3DEKernel via a DLL interface in `O3DEKernel.dll` (Windows), `libO3DEKernel.so` (Linux/Android) or `libO3DEKernel.dylib` (MacOS/iOS) to enforce the use of the one `AZ::Environment` (see translation of ancient inscription).
```

Reply:
We should leave the ancient inscription part out of it.

Reply:
```suggestion
Only one [AZ::Environnment](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Environment.h#L114-L115) instance exists and is accessed through the O3DEKernel via a DLL interface in `O3DEKernel.dll` (Windows), `libO3DEKernel.so` (Linux/Android) or `libO3DEKernel.dylib` (MacOS/iOS) to enforce the use of the one `AZ::Environment`.```

Reply:
If you don't want this documentation on the New York Times best seller list...